generator/metaline_builder.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
```

Comment:
No. feature_builder, restriction_collector. Either way is good, unless there is a really good reason.